### ISSUE TITLE
Resource command

### ISSUE THREAD

**Issue by InbalZilberman** [cyberark/conjur-api-python3#215]
## **Feature Overview & Customer Need**

**As a**  Conjur user 
**I would like to** use the Conjur cli 
**In order to** understand the status of Conjur resources

**As a**  Conjur user 
**I would like to** use the Conjur cli 
**In order to** understand which roles have what privileges on Conjur's resources

-----DRAFT -----
A new command will be called resource
We will use 2 sub commands
* exists - Determines whether a resource exists
* permitted_roles - List roles with a specified privilege on the resource

The commands will be used as follow:

TBD

Args description:

-r --resource <Resource identifier> - provide the resource identifier

Expected behavior should not change from the corresponding command in Ruby CLI 
@sharonr78 should provide the right help screen
Help is according to https://ljfz3b.axshare.com/#id=s9nycf&p=conjur_help__policy&g=1

## Quality

Make sure we have test coverage of the resource command.  Create test plan and execute accordingly.

## Process logic and Demo

Conjur user with a machine that the Python CLI already been installed and conjur init & login run against a Conjur/ Conjur Enterprise machine with the following resources:
TBD

Demo each option that was implemented
Show the resource exists show that these resources are found
Show that the resources has permitted_roles correct

Delete a resource by using !delete in a policy
Show that the deleted resource does not  exists yet others are found
Show that the option of running permitted_roles is failing.

## User messages

All user messages regarding resource actions should be reviewed

Especially error messages 

if an argument is missing we need to return the help of the command 

## **Documentation**

Please provide enhance documentation in online help and readme

## DOD

- [ ] Implement the resource's commands Conjur Python CLI in all 3 platforms
- [ ] Demo the flow described above
- [ ] Automatic integration tests written according to a test plan and passed successfully
- [ ] Fill in the user messages - link to a page with all user messages for CLI saved and reviewed. 
- [ ] Fill in the logs and place here a link to the logs files - link to a page with all logs for CLI saved and reviewed. 
- [ ] Logs were reviewed by TW and PO
- [ ] Are there new Audits to this feature? If so please document them and provide here a link (N/A)

**Comment by szamir1** [comment]
@InbalZilberman 
added the following under the logic section:

given The following resources are configured:
MyAccount:group:conjur-root-admins

The user wants to check if the resource exists,
conjur resource exists -r MyAccount:group:test-group
false


conjur resource permitted_roles - i MyAccount:group:conjur-root-admins -p read
[
  "MyAccount:user:admin"
]


If the user enters a non-existing privileged role, we need to return the help of the command
For example: 
conjur resource permitted_roles -r MyAccount:group:conjur-root-admins -p delete

**Comment by InbalZilberman** [comment]
Done thank you!!

**Comment by sharonr78** [comment]
@InbalZilberman @szamir1 

1. Same as in "role" command, I suggest to replace "exists" with "**check**" or "**verify**" since we're checking/querying/verifying whether that resource exists. The true/false response still works here. 
2. As for "permitted_roles", I also suggest to change this sub-command to "list_roles" since it's a verb. I'm not sure we need to specifically use "permitted". We'll have the explanation in the help screen.
WDYT?

**Comment by InbalZilberman** [comment]
> @InbalZilberman @szamir1
> 
> 1. Same as in "role" command, I suggest to replace "exists" with "**check**" or "**verify**" since we're checking/querying/verifying whether that resource exists. The true/false response still works here.
> 2. As for "permitted_roles", I also suggest to change this sub-command to "list_roles" since it's a verb. I'm not sure we need to specifically use "permitted". We'll have the explanation in the help screen.
>    WDYT?

I agree with # 2 but regarding exists this is a verb and is much more intuitive than verify. Let's change it from exists to exist.

**Comment by sharonr78** [comment]
> I agree with # 2 but regarding exists this is a verb and is much more intuitive than verify.

OK thanks Inbal, then we're agreeing that the subcommand for both "Role" and "Resource" commands will be "exist", e.g.,
`conjur resource exist -i "demo:host:ansible/ansible-01"`

**Comment by InbalZilberman** [comment]
I think it should be exists as resource is an it

**Comment by sigalsax** [comment]
I am not sure why we even need `exists` because the user can just run `conjur list` and see the actual item in the server

`conjur resource permitted_roles` should be `conjur resource permitted-roles` to follow suite with the rest of the commands like `rotate-api-key` and `change-password`

@InbalZilberman

**Comment by sigalsax** [comment]
@InbalZilberman above it says
> list-roles - List roles with a specified privilege on the resource

but in the following section it references permitted_roles. Which one is it?

**Comment by sigalsax** [comment]
Would be helpful to also link the supporting REST for more detailed info. For example this command's REST is https://docs.conjur.org/Latest/en/Content/Developer/Conjur_API_Check_Permission.htm?tocpath=Developer%7CREST%C2%A0APIs%7C_____21

**Comment by InbalZilberman** [comment]
> @InbalZilberman above it says
> 
> > list-roles - List roles with a specified privilege on the resource
> 
> but in the following section it references permitted_roles. Which one is it?

Fixed, thanks

**Comment by InbalZilberman** [comment]
> I am not sure why we even need `exists` because the user can just run `conjur list` and see the actual item in the server
> 
> `conjur resource permitted_roles` should be `conjur resource permitted-roles` to follow suite with the rest of the commands like `rotate-api-key` and `change-password`
> 
> @InbalZilberman

@sigalsax I have fixed the doc to list-roles. But now wonder if list-permitted-roles isnt better WDYT? it is long but maybe more comprehensive?

**Comment by sigalsax** [comment]
I like the idea but I do think it is too long. I think `list-roles` or `show-roles` (@sharonr78 ) would suffice here because we have the command and a follow up `read` `execute` `write` flag so it is clear that together we want to see all roles that have read/execute/write on the variable we pass in
